src/amltk/exceptions.py

```diff
@@ -110,3 +110,50 @@ def __str__(self) -> str:
             f"Duplicate names found in {self.node.name} and can't be handled."
             f"\nnodes: {[n.name for n in self.node.nodes]}."
         )
+
+
+class MatchDimensionsError(Exception):
+    """An exception raised for errors related to matching dimensions in a pipeline."""
+
+    def __init__(self, layer_name: str, param: str | None) -> None:
```

Comment:
Just as a note, the expected definition of an Exception is `def __init__(*args: Any)`. While we can be more specific here, it sort of breaks a normal contract of implmeneting exceptions. Don't really mind here, just letting you know.

```python
class MatchDimenionError(Exception):
    def __init__(self, layer_name: str, param: str | None, *args: Any) -> None:
    	super.__init__(..., *args)
```